From b5a5e4cd039e0b874b022721503a161fef2b85a8 Mon Sep 17 00:00:00 2001
From: agaraix <anath.bandux.garai@intel.com>
Date: Sat, 1 Feb 2020 21:46:55 +0530
Subject: [PATCH] UGW_SW-47271 : PPA acceleration fix

Member extmark re-positioned in SKB structure definition and thus ensured its initialization.
---
 include/linux/skbuff.h | 6 +++---
 net/core/skbuff.c      | 3 ---
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index 099b4b7ac6d6..143490849c18 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -809,6 +809,9 @@ struct sk_buff {
 #if defined(CONFIG_LTQ_CBM) || IS_ENABLED(CONFIG_DIRECTCONNECT_DP_API)
 	__u32 DW0, DW1, DW2, DW3;
 #endif
+#ifdef CONFIG_NETWORK_EXTMARK
+	__u32		extmark;
+#endif
 #ifdef CONFIG_PPA
 	void			*ptr_ppa_pitem;
 #endif
@@ -824,9 +827,6 @@ struct sk_buff {
 				*data;
 	unsigned int		truesize;
 	atomic_t		users;
-#ifdef CONFIG_NETWORK_EXTMARK
-	__u32		extmark;
-#endif
 };
 
 #ifdef __KERNEL__
diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index eb33bee112c8..4ae451dd794d 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -895,9 +895,6 @@ static void __copy_skb_header(struct sk_buff *new, const struct sk_buff *old)
 	memcpy(&new->headers_start, &old->headers_start,
 	       offsetof(struct sk_buff, headers_end) -
 	       offsetof(struct sk_buff, headers_start));
-#ifdef CONFIG_NETWORK_EXTMARK
-	new->extmark	 = old->extmark;
-#endif
 #if IS_ENABLED(CONFIG_PPA)
 	if (ppa_hook_pitem_refcnt_inc_fn && new->ptr_ppa_pitem)
 		ppa_hook_pitem_refcnt_inc_fn(new);
